src/app/dialogs/node-versions/node-versions.dialog.ts

```diff
@@ -36,12 +46,39 @@ import { Store } from '@ngrx/store';
 })
 export class NodeVersionsDialogComponent {
   node: MinimalNodeEntryEntity;
+  file: File;
+  isTypeList = true;
 
-  constructor(@Inject(MAT_DIALOG_DATA) data: any, private store: Store<any>) {
+  /** Emitted when a version is restored or deleted. */
+  @Output()
+  refreshEvent: EventEmitter<Node> = new EventEmitter<Node>();
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA) data: any,
+    private store: Store<AppStore>,
+    private dialogRef: MatDialogRef<NodeVersionsDialogComponent>
+  ) {
     this.node = data.node;
+    this.file = data.file;
+    this.isTypeList = data.isTypeList !== undefined ? data.isTypeList : true;
   }
 
-  uploadError(errorMessage: string) {
+  onUploadError(errorMessage: string) {
     this.store.dispatch(new SnackbarErrorAction(errorMessage));
   }
+
+  handleUpload(ev) {
```

Comment:
try to avoid 1 char or 2 char variable name it doesn't help readability of the code and add the type